### ISSUE TITLE
Invert Enter key behaviour

### ISSUE THREAD

**Issue by lgrahl** [threema-ch/threema-web#274]
Text is being submitted by pressing the `Enter` key and adding a new line by pressing `Shift+Enter`. Some people may want to invert this behaviour.

(Request from https://threema-forum.de/thread-3299.html)

**Comment by rugk** [comment]
…so add an *option*…

**Comment by dbrgn** [comment]
I cannot imagine that the majority of people wants to use the mouse or a key combination to submit a text message when using Threema Web.

An option for this could be added to the settings dialog. This feature request has low priority for us, but pull requests are welcome.

**Comment by dbrgn** [comment]
Part 1: Add settings for inverted submit behavior.

- In the settings dialog (https://github.com/threema-ch/threema-web/blob/master/src/partials/dialog.settings.html), add a new section called "Composing Text"
- In that section, add two radio buttons:
  - Enter to send, Shift+Enter for newline (enabled by default)
  - Ctrl+Enter to send, Enter for newline
- The settings can be persisted directly in the [SettingsController](https://github.com/threema-ch/threema-web/blob/master/src/partials/messenger.ts#L132) (no need to create a separate service like the NotificationService). To read/write settings, use `settingsService.storeUntrustedKeyValuePair(...)` and `settingsService.retrieveUntrustedKeyValuePair(...)`.
- For storing the settings with the SettingsService, use the key "submitBehavior" and the value "enterSend" or "enterNewline"

Part 2: Implementing behavioral change.

- When a user presses enter, [this code](https://github.com/threema-ch/threema-web/blob/master/src/directives/compose_area.ts#L278-L303) is run.
- React to key inputs depending on the chosen setting (retrieve it using the SettingsService)

**Comment by Dreema** [comment]
"In that section, add two radio buttons:
Enter to send, Shift+Enter for newline (enabled by default)
Ctrl+Enter to send, Enter for newline"

This would be very good!

**Comment by andyggg** [comment]
I totally agree with lgrahl and Dreema: it happened several times to me to accidentally send a premature message when I actually intended to just add a line break. 
I'd very much appreciate to have an option invert the behaviour - just like the one that the iOS-Threreema-App already "internally" provides, too.

**Comment by lgrahl** [comment]
Please no more :+1: comments. Instead, use the button. :slightly_smiling_face:

**Comment by lgrahl** [comment]
FYI @dbrgn said this is blocked by the compose area refactoring.

**Comment by rkraneis** [comment]
Any news? I switch to threema web solely to compose longer answers that would be awkward to type on a phone (and already changed the sending behaviour there).
